aws/resource_aws_instance.go

```diff
@@ -105,10 +104,11 @@ func resourceAwsInstance() *schema.Resource {
 			},
 
 			"private_ip": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Computed: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.IsIPv4Address,
```

Comment:
This validation means that you cannot pass an empty string and as a result ALL of our EC2 instances that were created with a module which passes an empty string (which is a valid value) when an IP is not specified are now producing diffs.